src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -100,16 +89,8 @@ private async Task PackageInitializationBackgroundThreadAsync(PackageLoadTasks p
             isMainThreadTask: true,
```

Comment:
so, this goes to my feedback from before.  why are we initializing on the BG just to kick this tack over to the foreground?  why not just do this on the FG?   

(note: i'm not saying this is wrong.  i'm just saying i don't understand the 'why?' part fo this, and i really wish the code was doc'ed for future understanding).

Reply:
I think in this specific case, SettingsEditorFactory is fairly expensive to create via MEF. There's a more general piece of work to clean that up since really should be cheap to create and pass to RegsiterEditorFactory, but that's not how it's written right now.

Reply:
my ask is that we then doc these decisions *in the code*. :)